### PR TITLE
Throw exception on invalid color scheme name.

### PR DESCRIPTION
**Pull request by karjonas**

### CODE REVIEW

File: brayns/parameters/GeometryParameters.cpp [hunk 1]
```diff
@@ -119,13 +119,14 @@ GeometryParameters::GeometryParameters()
...
-        PARAM_GEOMETRY_QUALITY.c_str(), po::value<std::string>(),
-        "Geometry rendering quality [low|medium|high]")(
+        "[float]")(PARAM_COLOR_SCHEME.c_str(), po::value<std::string>(),
+                   "Color scheme to be applied to the geometry "
+                   "[none|neuron-by-id|neuron-by-type|neuron-by-segment-type|"
+                   "neuron-by-layer|neuron-by-mtype|neuron-by-etype|neuron-by-"
+                   "target|protein-by-id|protein-atoms|protein-chains|protein-"
+                   "residues]")(PARAM_GEOMETRY_QUALITY.c_str(),
+                                po::value<std::string>(),
+                                "Geometry rendering quality [low|medium|high]")(
```
Comment (hernando) [inline brayns/parameters/GeometryParameters.cpp]: The whole code block for adding options is unreadable. Can you do some hand formatting to all the options and add // clang-format off // clang-format surrounding it?

File: brayns/parameters/GeometryParameters.cpp [hunk 2]
```diff
@@ -211,10 +212,20 @@ void GeometryParameters::parse(const po::variables_map& vm)
...
+                if (colorScheme == COLOR_SCHEMES[i])
+                {
+                    _colorScheme = static_cast<ColorScheme>(i);
+                    matchedScheme = true;
+                    break;
+                }
+
+            if (!matchedScheme)
+                throw po::error("No match for color scheme '" + colorScheme);
+        }
```
Comment (hernando) [inline brayns/parameters/GeometryParameters.cpp]: Does it look shorter with std::find? (in fact, that would be a reason for using std::array)

**Comment by rolandjitsu** [comment]
This is legacy and will/should be moved to a plugin eventually. Not sure if it's worth spending any more time on it.

File: brayns/parameters/GeometryParameters.cpp [hunk 3]
```diff
@@ -71,18 +71,18 @@ const std::string PARAM_MORPHOLOGY_USE_SDF_GEOMETRIES =
...
-                                       "neuron-by-segment-type",
-                                       "neuron-by-layer",
-                                       "neuron-by-mtype",
-                                       "neuron-by-etype",
-                                       "neuron-by-target",
-                                       "protein-by-id",
-                                       "protein-atoms",
-                                       "protein-chains",
-                                       "protein-residues"};
+const std::array<std::string, 12> COLOR_SCHEMES = {"none",
```
[the remark this answers is not in the document]
Comment (karjonas) [reply brayns/parameters/GeometryParameters.cpp]: That was not possible.

File: brayns/parameters/GeometryParameters.cpp (hunk 2 above)
Comment (karjonas) [reply brayns/parameters/GeometryParameters.cpp]: done

File: brayns/parameters/GeometryParameters.cpp (hunk 1 above)
Comment (karjonas) [reply brayns/parameters/GeometryParameters.cpp]: done

File: brayns/parameters/GeometryParameters.cpp [hunk 4]
```diff
@@ -71,18 +71,18 @@ const std::string PARAM_MORPHOLOGY_USE_SDF_GEOMETRIES =
...
+                                                   "neuron-by-type",
+                                                   "neuron-by-segment-type",
+                                                   "neuron-by-layer",
+                                                   "neuron-by-mtype",
+                                                   "neuron-by-etype",
+                                                   "neuron-by-target",
+                                                   "protein-by-id",
+                                                   "protein-atoms",
+                                                   "protein-chains",
+                                                   "protein-residues"};
```
[the remark this answers is not in the document]
Comment (karjonas) [reply brayns/parameters/GeometryParameters.cpp]: I will leave it like this until it is removed.

[the remark this answers is not in the document]
Comment (hernando) [reply brayns/parameters/GeometryParameters.cpp]: Sure. I was just commenting that we will need to change it eventually.

**Pull request merged**